train.py

```diff
@@ -259,59 +272,139 @@ def run_train(
                 plot = cfg.logging.plots_class.plot_batch(batch=batch, cfg=cfg)
                 log_plot(cfg, plot, "train_data")
 
-            # Forward pass
-            if cfg.environment.mixed_precision:
-                with autocast():
-                    output_dict = model.forward(batch)
-            else:
-                output_dict = model.forward(batch)
-
-            loss = output_dict["loss"]
-            if ~np.isfinite(loss.item()) and (num_updates > 20):
-                raise LLMTrainingException(
-                    "NaN caught in loss during training. "
-                    "Please, reduce learning rate, change dtype, "
-                    "or disable mixed precision. "
-                    "Alternatively, gradient clipping may help to stabilize training."
+            if cfg.training.use_rlhf:
+                torch.inference_mode(mode=True)
+                logger.debug("Rollout: Generating response from active model")
+                output_dict = {}
+                output_dict["predicted_answer_ids"] = model.generate(
+                    batch, model.cfg, remove_prompt=True
```

Comment:
I noticed that I mixed up `max_length_answer` with min length and max length for inference (causing OOMs).